spec/jobs/browse_everything_ingest_job_spec.rb

```diff
@@ -11,4 +12,133 @@
+        url: "https://www.googleapis.com/drive/v3/files/1XzEWzaluDQj0qIH4xqcgxQXcr6?alt=media",
+        file_size: 0,
+        auth_token: "ya29.GltkBysb2tnNUmUscg0xL1BrXe3g-vALh4Xo5fXrPfOSpJLaPNxfxThztNi8Qdl3uBkEomFis8HeLDpsYb3tKMH1Kf6Zsih63J8HN0KowPHUbVM__",
+        auth_header: JSON.generate("Authorization" => "Bearer ya29.GltkBysb2tnNUmUscg0xL1BrXe3g-vALh4Xo5fXrPfOSpJLaPNxfxThztNi8Qdl3uBkEomFis8HeLDpsYb3tKMH1Kf6Zsih63J8HN0KowPHUbVM__"),
+        type: "application/x-directory",
+        provider: "google_drive"
+      }
+    end
+    let(:pending_upload) do
+      PendingUpload.new(pending_upload_attributes)
+    end
+    let(:pending_uploads) { [pending_upload] }
+    let(:pending_upload_ids) { pending_uploads.map(&:id).map(&:to_s) }
+    let(:adapter) { Valkyrie::MetadataAdapter.find(:indexing_persister) }
+    let(:storage_adapter) { Valkyrie::StorageAdapter.find(:disk_via_copy) }
+    let(:change_set_persister) do
+      ChangeSetPersister.new(metadata_adapter: adapter, storage_adapter: storage_adapter)
+    end
+    let(:provider) { instance_double(BrowseEverything::Driver::GoogleDrive) }
+    let(:provider_class) { class_double(BrowseEverything::Driver::GoogleDrive) }
+    let(:retriever) { instance_double(BrowseEverything::Retriever) }
+    let(:auth_token) do
+      "ya29.GltkBysb2tnNUmUscg0xL1BrXe3g-vALh4Xo5fXrPfOSpJLaPNxfxThztNi8Qdl3uBkEomFis8HeLDpsYb3tKMH1Kf6Zsih63J8HN0KowPHUbVM__"
+    end
+    let(:auth_header) do
+      JSON.generate("Authorization" => "Bearer #{auth_token}")
+    end
+    let(:member_resources) do
+      [
+        {
+          id: "1v3gRwJBqxuO4dU1rLHhwCUTTj-Q6XSdN",
+          url: "https://www.googleapis.com/drive/v3/files/1v3gRwJBqxuO4dU1rLHhwCUTTj?alt=media",
+          auth_token: auth_token,
+          auth_header: auth_header,
+          file_name: "cea.tif",
+          file_size: 270_993,
+          container: false,
+          provider: "google_drive"
+        }
+      ]
+    end
+
+    before do
+      change_set = DynamicChangeSet.new(scanned_resource)
+      change_set.validate(pending_uploads: pending_uploads)
+      change_set_persister.save(change_set: change_set)
+
+      allow(provider_class).to receive(:authorization_header).and_return(auth_header)
+      allow(provider).to receive(:class).and_return(provider_class)
+      allow(BrowseEverything::Retriever).to receive(:build_provider).and_return(provider)
+      allow(BrowseEverything::Retriever).to receive(:new).and_return(retriever)
+      allow(retriever).to receive(:member_resources).and_return(member_resources)
+      allow(retriever).to receive(:download).and_return(fixture_file_upload("files/example.tif", "image/tiff"))
+    end
+
+    it "recurses through and ingests the member resources" do
+      described_class.perform_now(scanned_resource.id.to_s, "ScannedResourcesController", pending_upload_ids)
+      updated = adapter.query_service.find_by(id: scanned_resource.id)
+      expect(updated.decorate.file_sets).not_to be_empty
```

Comment:
can you do more than one file and check for the right number of file sets? I think that will make the test slightly more robust.

Reply:
https://github.com/pulibrary/figgy/pull/3290/files#diff-6e1d982e3b667f79274c6a15bda715b3R96 now checks for the number of FileSets.